PostFreeSurfer/PostFreesurferPipelineLongPrep.sh

```diff
@@ -0,0 +1,434 @@
+# The next block computes the transform from T1w_acpc_dc (cross) to T1w_acpc_dc (long_template).
+Timepoint_long=$Timepoint_cross.long.$Template
+T1w_dir_cross=$StudyFolder/$Timepoint_cross/T1w
+T1w_dir_long=$StudyFolder/$Timepoint_long/T1w
+T1w_dir_template=$StudyFolder/$Subject.long.$Template/T1w
+LTA_norm_to_template=$T1w_dir_template/$Template/mri/transforms/${Timepoint_cross}_to_${Template}.lta
+
+if (( TemplateProcessing == 0 )); then #timepoint mode
+
+    log_Msg "Timepoint processing: computing the transform from T1w_acpc_dc (cross) to T1w_acpc_dc (longitudinal template)"
+    mkdir -p $T1w_dir_long/xfms
+    #Snorm=$T1w_dir_long/$Timepoint_long/mri/norm
+    Snorm=$T1w_dir_cross/$Timepoint_cross/mri/norm
+    mri_convert $Snorm.mgz $Snorm.nii.gz
+
+    T1w_cross=$T1w_dir_cross/T1w_acpc_dc_restore.nii.gz
+    T1w_long=$T1w_dir_long/T1w_acpc_dc_restore.nii.gz
+    T2w_long=$T1w_dir_long/T2w_acpc_dc_restore.nii.gz
+
+    #1. create resample transform from timepoints' T1w to 'norm' (equiv. to 'orig') space. This only applies reorient/resample, no actual registration
+    lta_convert --inlta identity.nofile --src $T1w_cross --trg $Snorm.nii.gz --outlta $T1w_dir_long/xfms/T1w_cross_to_norm.lta	
+
+    #2. concatnate previous transform with TP(orig)->template(orig) transform computed by FS.
+    mri_concatenate_lta $T1w_dir_long/xfms/T1w_cross_to_norm.lta $LTA_norm_to_template $T1w_dir_long/xfms/T1w_cross_to_template.lta
+    #3. Invert TP->norm transforms. Again, this is only reorient/resample, this isn't an actual registration transform.
+    lta_convert --inlta $T1w_dir_long/xfms/T1w_cross_to_norm.lta --outlta $T1w_dir_long/xfms/norm_to_T1w_cross.lta --invert
+    #4. Combine TP->template and orig->TP transforms to get TP->TP(HCP template) transform.
+    mri_concatenate_lta $T1w_dir_long/xfms/T1w_cross_to_template.lta $T1w_dir_long/xfms/norm_to_T1w_cross.lta $T1w_dir_long/xfms/T1w_cross_to_T1w_long.lta	
+    #5. convert the final TP_T1w->TP_T1w(HCP template) LTA transform to .mat(FSL).
+    lta_convert --inlta $T1w_dir_long/xfms/T1w_cross_to_T1w_long.lta --src $T1w_cross --trg $T1w_cross --outfsl $T1w_dir_long/xfms/T1w_cross_to_T1w_long.mat
+else #Template mode
+    :
+fi    
+#end block
+
+#################################################################################################################
+# The next block resamples T1w from native to the longitudinal acpc_dc template space, using cross-sectional 
+# readout distortion correction warp.
+
+if (( TemplateProcessing == 0 )); then 
+    log_Msg "Timepoint $Timepoint_long: combining warps and resampling T1w_acpc_dc to longitudinal template space"
+    T1w_cross_gdc_warp=$T1w_dir_cross/xfms/${T1wImage}1_gdc_warp.nii.gz
+    if [ ! -f "$T1w_cross_gdc_warp" ]; then
+        log_Msg "NOT USING GRADIENT DISTORTION CORRECTION WARP"
+        create_zero_warp $T1w_dir_cross/${T1wImage}1_gdc.nii.gz $T1w_cross_gdc_warp
```

Comment:
Where does this get used?  Does it really need to exist when gdc isn't done?

Reply:
This looks like a trace of old confusion I thought is gone. Indeed, gradient distortion correction is done in PreFreesurfer before any transforms and therefore need not be handled in PostFreeSurferPipelineLongPrep.